### PR TITLE
Add domain option to onepassword lookup

### LINKED ISSUE #2734: onepassword lookup does not work for 1password.eu
### Summary

When using the community.general.onepassword lookup, the module assumes a subdomain of 1password.com, so tasks using subdomains of 1password.eu fail.

### Issue Type

Bug Report

### Component Name

https://github.com/ansible-collections/community.general/blob/f74b83663bcc1f6269a2bb56ba646f24f6218578/plugins/lookup/onepassword.py#L171

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.11.1] 
  config file = None
  configured module search path = ['/home/bendwyer/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/venv/ansible/lib64/python3.6/site-packages/ansible
  ansible collection location = /home/bendwyer/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/venv/ansible/bin/ansible
  python version = 3.6.8 (default, Aug 24 2020, 17:57:11) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
  jinja version = 2.11.3
  libyaml = True
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
CACHE_PLUGIN(env: ANSIBLE_CACHE_PLUGIN) = jsonfile
CACHE_PLUGIN_CONNECTION(env: ANSIBLE_CACHE_PLUGIN_CONNECTION) = /home/bendwyer/.ansible/cache/fact_cache
CACHE_PLUGIN_TIMEOUT(env: ANSIBLE_CACHE_PLUGIN_TIMEOUT) = 7200
DEFAULT_FORKS(env: ANSIBLE_FORKS) = 50
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/home/bendwyer/automate/ansible/inventory/environments/production']
INVENTORY_CACHE_ENABLED(env: ANSIBLE_INVENTORY_CACHE) = True
INVENTORY_CACHE_PLUGIN(env: ANSIBLE_INVENTORY_CACHE_PLUGIN) = jsonfile
INVENTORY_CACHE_PLUGIN_CONNECTION(env: ANSIBLE_INVENTORY_CACHE_CONNECTION) = /home/bendwyer/.ansible/cache/inventory_cache
INVENTORY_CACHE_TIMEOUT(env: ANSIBLE_INVENTORY_CACHE_TIMEOUT) = 7200
```

### OS / Environment

```console
$ cat /etc/centos-release
CentOS Linux release 8.3.2011
```

### Steps to Reproduce

```yaml (paste below)
---
- name: test
  hosts: localhost
  connection: local
  gather_facts: false
  vars:
    record_name:  # name of item we want the password for
    subdomain:  # 1Password subdomain (currently only for 1Password.com)
    master_password:  # password for logging into 1Password vault
    username:  # email for logging into 1Password vault
    secret_key:  # secret key for logging into 1Password vault

  tasks:
    - name: Retrieve password when never signed in to 1Password
      ansible.builtin.debug:
        var: lookup("community.general.onepassword", "{{ record_name }}", subdomain="{{ subdomain }}", master_password="{{ master_password }}", username="{{ email }}", secret_key="{{ secret_key }}")
    
```

### Expected Results

Login to 1password.eu and retrieve password associated with `{{ record_name }}`.

### Actual Results

```console (paste below)
TASK [Retrieve password when never signed in to 1Password] *********************
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'community.general.onepassword'. Error was a <class 'ansible.errors.AnsibleLookupError'>, original message: Your Secret Key has changed. You must enter your new Secret Key to sign in.\nEnter the Secret Key for [REDACTED - username] at [REDACTED - subdomain].1password.com: [ERROR] 2021/06/07 10:42:26 invalid account key format length 0\n"}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes #2734.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/lookup/onepassword.py

### CODE REVIEW

**Comment by aminvakil** [comment]
@samdoran Do you think it would be necessary to use FQDN instead of `subdomain`.`domain`, so if one day :) 1password lets users using their own domain, we support that too?
Or as @felixfontein said: "or mypersonal1password.com which you registered and CNAME'd (or whatever) to bla.1password.com" ?

Or is this PR ok as it is?

File: plugins/lookup/onepassword.py [hunk 1]
```diff
@@ -30,6 +30,10 @@
         aliases: ['vault_password']
       section:
         description: Item section containing the field to retrieve (case-insensitive). If absent will return first match from any section.
+      domain:
+        description: Domain of 1Password. Default is U(1password.com).
+        version_added: 3.2.0
+        default: '1password.com'
```
Comment (felixfontein) [inline plugins/lookup/onepassword.py:36]:
```suggestion
        default: '1password.com'
        type: str
```
(The type should always be mentioned in docs.)

File: plugins/lookup/onepassword.py [hunk 2]
```diff
@@ -265,6 +269,7 @@ def run(self, terms, variables=None, **kwargs):
         section = kwargs.get('section')
         vault = kwargs.get('vault')
         op.subdomain = kwargs.get('subdomain')
+        op.domain = kwargs.get('domain', '1password.com')
```
Comment (felixfontein) [inline plugins/lookup/onepassword.py:272]: Looks like this should be converted to using the config manager (`set_options`/`get_option`) eventually :)

**Review by felixfontein — COMMENTED** [review]
I'm happy with the current form, the idea of being able to specify the complete domain name is probably not needed.

**Comment by aminvakil** [comment]
@bendwyer Would you like to test this PR to see if it fixes your problem?

**Comment by bendwyer** [comment]
Hi @aminvakil, I have replied back in the issue thread:  https://github.com/ansible-collections/community.general/issues/2734#issuecomment-856567228

**Review by bendwyer — APPROVED** [review]
I have tested this PR on my own and it works for me! A massive thanks to @aminvakil for the rapid response and fix. 🍻

**Comment by aminvakil** [comment]
@bendwyer Thanks for testing this! This will be included in [3.2.0](https://github.com/ansible-collections/community.general/issues/582#issuecomment-852422621) which is due to be released in a couple of hours.
@felixfontein Thanks for reviewing and merging!

**Pull request merged**